### PR TITLE
Viz EC2 Services to Lambda - Part 1 - High Water Probability

### LINKED ISSUE #468: Move High Water Probability (and maybe ROF Probability) from EC2 to Lambda
With Corey's enhancement to bring the lambda max values into the step function, we can now easily have that same lambda function run high water probability services, and potentially ROF probability (if run times can be consistently lower than 15 minutes).

- [x] SRF High Water Probability
- [x] MRF High Water Probability
- [ ] SRF ROF Probability
- [ ] MRF ROF Probability

Dev Tasks
- [x] Update lambda_shared_funcs to support multiple range_tokens in file_pattern
- [x] Tweak max_values lambda configuration to support larger ephemeral storage, and viz db connection info (and rename?)
- [x] Add Probability python code to max_values lambda function
- [x] Disable/remove corresponding EC2 services

### PR DESCRIPTION
**Pull request by TylerSchrag-NOAA**
This PR migrates the ensemble-based High Water Probability products from the Viz EC2 to the Viz Max Values lambda function (now called Python Preprocessing). It broadly encapsulates the following changes:

**Changes**
- Renaming of the Viz Max Values lambda function to Viz Python Preprocessing
- Renaming of max_values (and variations) to python_preprocessing throughout code base
- Abstraction of python_preprocessing to function to include multiple "products", with max_values now being one of those products.
- New high_water_probability product script that supports both 12 hour SRF and 5 day GFS MRF configurations.
- Removal of high_water_probability pipeline and product file in the source-aws_loosa library
- Minor changes to viz_shared_functions to support multiple range tokens in file patterns

**Testing** - I've tested this new functionality with SRF, MRF, and ANA configurations fairly vigorously in the dev historic event version of the pipelines, which was created in August 2023. That said, I have not tested it with coastal configurations, which do rely on max_values for FIM preprocessing (although I don't expect my changes to affect these at all). I have also not thoroughly tested the actual map service component on the EGIS, as I cannot access the portal. I believe it's close enough to deploy to TI for testing this functionality.

**Deployment Considerations:**
- We should definitely wait until the upcoming public release deployment is well and settled before merging this into TI, as it includes foundational changes to the pipelines that we do not want to contend with for hot fixes.
- I changed the mrf service name to include "max", to be consistent with the SRF services. Let me know if that isn't correct, but we'll need to coordinate with WPOD on app changes for that.
- We will need to include a ingest schema db dump when deploying to UAT.

### CODE REVIEW

**Review by shawncrawley — APPROVED** [review]
This looks good - and it is super cool to be moving that much closer to no more EC2 pipeline processing! Well done.

**Review by CoreyKrewson-NOAA — CHANGES_REQUESTED** [review]

File: Core/LAMBDA/viz_functions/viz_initialize_pipeline/lambda_function.py [hunk 1]
```diff
@@ -602,30 +604,30 @@ def get_product_metadata(self, specific_products=None, run_only=True):
...
+            product['python_preprocesing_dependent'] = False
             if product.get('db_max_flows'):
                 self.db_max_flows.extend([max_flow for max_flow in product['db_max_flows'] if max_flow not in self.db_max_flows])
                 
-            if product.get('lambda_max_flows'):
-                product['lambda_max_flow_dependent'] = True
-                self.lambda_max_flows.extend([max_flow for max_flow in product['lambda_max_flows'] if max_flow not in self.lambda_max_flows])
+            if product.get('python_preprocessing'):
+                product['python_preprocesing_dependent'] = True
+                self.python_preprocessing.extend([python_preprocess for python_preprocess in product['python_preprocessing'] if python_preprocess not in self.python_preprocessing])
```
[the remark this answers is not in the document]
Comment (shawncrawley) [reply Core/LAMBDA/viz_functions/viz_initialize_pipeline/lambda_function.py:617]: I feel like I like it. I'm always on the fence a bit myself around these same exact configuration questions. If we add a new configuration option, I realize that it's annoying to have to spell out every pre-existing use of the now-option just for the sake of the one-off that the option was now created for. But you do always lose a bit of transparency with under the cover defaults. But I still think I lean toward the defaults existing so the option does not need to be spelled out everywhere.

[the remark this answers is not in the document]
Comment (CoreyKrewson-NOAA) [reply Core/LAMBDA/viz_functions/viz_initialize_pipeline/lambda_function.py:617]: yeah I go back and forth as well. I do think we need to chose one and stick with it for consistency. I lean towards using defaults but making sure we have a template that is up to date and clear

[the remark this answers is not in the document]
Comment (shawncrawley) [reply Core/LAMBDA/viz_functions/viz_initialize_pipeline/lambda_function.py:617]: Agreed. Especially on the well-documented template.

[the remark this answers is not in the document]
Comment (CoreyKrewson-NOAA) [reply Core/LAMBDA/viz_functions/viz_initialize_pipeline/lambda_function.py:617]: I realized that some other python_preprocessing args are technically (or could be) optional but we require it. I am just going to leave this as is for now and keep it required to stay consistent

**Review by CoreyKrewson-NOAA — APPROVED** [review]

**Pull request merged**